Fix build with Poppler 0.73.

This is an amalgamation of these upstream commits:
https://github.com/scribusproject/scribus/commit/d34e59bfe495250ba023ba0f99e672ee32300a27
https://github.com/scribusproject/scribus/commit/c43a89030026f8ffab5070b6935daee8bd74838a

diff --git a/scribus/plugins/import/pdf/slaoutput.h b/scribus/plugins/import/pdf/slaoutput.h
index c46448b80a..4dac7c3983 100644
--- a/scribus/plugins/import/pdf/slaoutput.h
+++ b/scribus/plugins/import/pdf/slaoutput.h
@@ -28,7 +28,9 @@ for which a new license (GPL+exception) is in place.
 #include "selection.h"
 #include "vgradient.h"
 
+#if POPPLER_ENCODED_VERSION < POPPLER_VERSION_ENCODE(0, 73, 0)
 #include <poppler/goo/gtypes.h>
+#endif
 #include <poppler/Object.h>
 #include <poppler/OutputDev.h>
 #include <poppler/Gfx.h>
diff --git a/scribus/plugins/import/pdf/importpdfconfig.h b/scribus/plugins/import/pdf/importpdfconfig.h
index f6b3950377..80ac796501 100644
--- a/scribus/plugins/import/pdf/importpdfconfig.h
+++ b/scribus/plugins/import/pdf/importpdfconfig.h
@@ -37,4 +37,11 @@ for which a new license (GPL+exception) is in place.
 #define getCString  c_str
 #endif
 
+#if POPPLER_ENCODED_VERSION >= POPPLER_VERSION_ENCODE(0, 73, 0)
+#define Guchar  unsigned char
+#define Gushort unsigned short
+#define Guint   unsigned int
+#define Gulong  unsigned long
+#endif
+
 #endif
